docs_src/microservices/security/Ch-SSH-Tunneling-HowToSecureDeviceServices.md

```diff
@@ -0,0 +1,297 @@
+# Security for EdgeX Stack
+
+This page describes one of options to secure the EdgeX software stack with running remote device services like device-virtual, device-rest, device-mqtt, and so on, via secure two-way SSH-tunnelings.
+
+## Basic SSH-Tunneling
+
+In this option to secure the EdgeX software stack, SSH tunneling is utilized. The basic idea is to create a secure SSH connection between  a local machine, the primary host, and a remote machine, the secondary, in which some micro-services or applications can be relayed.   In this particular example, the primary host is  running the whole EdgeX core services including core services and security services but **without** any device service.  The device services are running in the secondary or the remote machine.  
+
+The communication is secure because SSH port forwarding connection is encrypted by default.
+
+The SSH communication is established by introducing some extra SSH-related services:
+
+1) device-ssh-proxy: this is the service with ssh client opening up the SSH communication between the primary and the secondary
+
+2) device-ssh-remote: this is actually the SSH server or daemon service together with device services running on the remote machine
+
+The high-level diagram is shown as follows:
+
+![image](ssh-tunneling_device.png)
+
+"Top level diagram for SSH tunneling for device services"
+
+In the primary, the SSH tunneling initiated by device-ssh-proxy service to the remote running device services.  The dependencies that remote device services needed are reversely tunneling from the primary machine.
```

Comment:
The first sentence doesn't read well, not sure what it is trying to say. 

change primary to local/local machine

Reply:
reworded and changed